format_test.go

```diff
@@ -52,6 +52,11 @@ var formatTests = []struct {
 		i: pointer(30),
 		s: pointer(0),
 	}, "last", 1551342600},
+	{"american", formatsMap["american"], &result{
```

Comment:
i am unable to write a correct test for this format, could you please help me explain it out